libs/lib_parsing_tree_sitter/Parse_tree_sitter_helpers.ml

```diff
@@ -59,28 +59,33 @@ let line_col_to_pos file =
           done
         with
         | End_of_file ->
-            (* bugfix: this is wrong:  Hashtbl.add h (!line, 0) !charpos;
-             * because an ident on the last line would get
-             * the last charpos.
-             *)
-            ()
+            (* We need to add this in case there is a trailing \n in the
```

Comment:
So we are not only reporting errors properly but we're also fixing a bug?

Reply:
I should have addressed this better. Here's what happened:

1. I made it so that we don't silently skip when we can't calculate a charpos
2. This made our pattern parsing fail
3. This also made target parsing fail in a few instances
4. To make tests pass after the change in point 1, I made changes to address those failures

So, the existing automated tests cover the bug fixes I made.

This is a case where it's not great that I rebased---if I hadn't I could easily find you the github actions to show you which tests failed.

Reply:
Updated the PR comment

Reply:
Also clarified that I mean the dockerfile in the issue 😅